src/aw-client.ts

```diff
@@ -44,130 +54,182 @@ class AWClient {
         }
     } = {};
 
-    constructor(clientname: string, testing: boolean, baseurl: string | undefined = undefined) {
+    constructor(clientname: string, options?: {testing?: boolean, baseURL?: string}) {
```

Comment:
I think this is a good change, see: https://stackoverflow.com/questions/12826977/multiple-arguments-vs-options-object

Reply:
Aha, so it's rather that the limitation of not being of able to use kwargs in JS can make it hard to read since you cannot annotate the args in a nice manner. In that case it seems sane.

Reply:
yeah, as a rule of thumb, if you have more than 2 arguments (especially if they are optional), a `options` object will be more readable